Mark the rtree as loaded during deserialization

davisp opened the pull request:
A simple oversight gets a simple fix.

---
TYPE: IMPROVEMENT
DESC: Mark the rtree as loaded during deserialization

davisp commented:
Whoa. That made the test suite surprisingly angry.

ypatia commented:
So, I think I need some context here on why you needed to do this change.
There is something very ugly around `loaded_metadata.rtree_`, our classes not being C41 compliant an all, and I most certainly take the blame on not documenting the ugliness in a ticket when I hit it, but check this out :
- https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/serialization/array.cc#L312
and 
- https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/serialization/fragment_info.cc#L252

I can't remember from the top of my head why I needed to go that way, but I am pretty sure I hit the same failures as you and found an explanation that I worked around that (poor?) way. I'll investigate a bit more and let you know. Sorry for the confusion.

davisp commented:
@ypatia So I'm not sure I actually do. This was just something I stumbled across while trying to update the memory tracking code.

The way I was going was to register all allocations with a tracking object and I added some logic to reject any allocations that had already happened since that seemed like any instance of that would be a bug. This ended up catching the reloading of the rtree in FragmentMetadata after deserialization. I didn't think too hard when I added this change thinking it was just an oversight, but obviously doing this causes something in the reader logic.

I spent a few hours yesterday looking at this and I didn't find the underlying issue. I'll catch up with @KiterLuc later today to see what he thinks, but I'm guessing I'll probably relax my allocation requirements and file a story to follow up on this later.
